File: app/src/test/java/com/annushkaproject/programmerscalculator/StandardTests/StandardDivisionTest.java
package com.annushkaproject.programmerscalculator.StandardTests;

import android.database.sqlite.SQLiteBlobTooBigException;

import com.annushkaproject.programmerscalculator.model.CalculationModel;
import com.annushkaproject.programmerscalculator.model.Value;
import com.annushkaproject.programmerscalculator.utils.StandardOperationsUtil;

import org.junit.Test;

import java.math.BigDecimal;

import static com.annushkaproject.programmerscalculator.model.Operator.DIVIDE;
import static org.junit.Assert.assertEquals;

public class StandardDivisionTest {
    @Test
    public void testIwoIntegers() {
        BigDecimal firstValue = BigDecimal.valueOf(25);
        BigDecimal secondValue = BigDecimal.valueOf(5);
        CalculationModel model = new CalculationModel(firstValue, secondValue, DIVIDE);
        assertEquals(5, StandardOperationsUtil.calculateResultForTwoSidedOperator(model), 0);
    }

    @Test
    public void testFirstFloat() {
        BigDecimal firstValue = BigDecimal.valueOf(-0.0001);
        BigDecimal secondValue = BigDecimal.valueOf(2);
        CalculationModel model = new CalculationModel(firstValue, secondValue, DIVIDE);
        assertEquals(-0.00005, StandardOperationsUtil.calculateResultForTwoSidedOperator(model), 0);
    }

    @Test
    public void testFloatResult() {
        BigDecimal firstValue = BigDecimal.valueOf(3);
        BigDecimal secondValue = BigDecimal.valueOf(2);
        CalculationModel model = new CalculationModel(firstValue, secondValue, DIVIDE);
        assertEquals(1.5, StandardOperationsUtil.calculateResultForTwoSidedOperator(model), 0);
    }

    @Test
    public void testFloatResultLessThanOne() {
        BigDecimal firstValue = BigDecimal.valueOf(1);
        BigDecimal secondValue = BigDecimal.valueOf(2);
        CalculationModel model = new CalculationModel(firstValue, secondValue, DIVIDE);
        assertEquals(0.5, StandardOperationsUtil.calculateResultForTwoSidedOperator(model), 0);
    }
}

File: app/src/test/java/com/annushkaproject/programmerscalculator/StandardTests/StandardDivisionTest.kt
package com.annushkaproject.programmerscalculator.StandardTests

import com.annushkaproject.programmerscalculator.model.CalculationModel
import com.annushkaproject.programmerscalculator.model.Operator
import com.annushkaproject.programmerscalculator.utils.StandardOperationsUtil.calculateResultForTwoSidedOperator
import org.junit.Assert
import org.junit.Test
import java.math.BigDecimal

class StandardDivisionTest {
    @Test
    fun testIwoIntegers() {
        val firstValue = BigDecimal.valueOf(25)
        val secondValue = BigDecimal.valueOf(5)
        val model = CalculationModel(firstValue, secondValue, Operator.remainder_divide)
        Assert.assertEquals(5.0, calculateResultForTwoSidedOperator(model), 0.0)
    }

    @Test
    fun testIwoIntegersRountResult() {
        val firstValue = BigDecimal.valueOf(5)
        val secondValue = BigDecimal.valueOf(9)
        val model = CalculationModel(firstValue, secondValue, Operator.remainder_divide)
        Assert.assertEquals(0.5555555555555556, calculateResultForTwoSidedOperator(model), 0.0)
    }

    @Test
    fun testFirstFloat() {
        val firstValue = BigDecimal.valueOf(-0.0001)
        val secondValue = BigDecimal.valueOf(2)
        val model = CalculationModel(firstValue, secondValue, Operator.remainder_divide)
        Assert.assertEquals(-0.00005, calculateResultForTwoSidedOperator(model), 0.0)
    }

    @Test
    fun testFloatResult() {
        val firstValue = BigDecimal.valueOf(3)
        val secondValue = BigDecimal.valueOf(2)
        val model = CalculationModel(firstValue, secondValue, Operator.remainder_divide)
        Assert.assertEquals(1.5, calculateResultForTwoSidedOperator(model), 0.0)
    }

    @Test
    fun testFloatResultLessThanOne() {
        val firstValue = BigDecimal.valueOf(1)
        val secondValue = BigDecimal.valueOf(2)
        val model = CalculationModel(firstValue, secondValue, Operator.remainder_divide)
        Assert.assertEquals(0.5, calculateResultForTwoSidedOperator(model), 0.0)
    }
}